utilities/backupable/backupable_db_test.cc

```diff
@@ -686,6 +692,30 @@ class BackupEngineTest : public testing::Test {
     backup_chroot_env_->DeleteFile(latest_backup_).PermitUncheckedError();
   }
 
+  void SetEnvsFromFileSystems() {
+    db_chroot_env_.reset(
+        new CompositeEnvWrapper(Env::Default(), db_chroot_fs_));
+    backup_chroot_env_.reset(
+        new CompositeEnvWrapper(Env::Default(), backup_chroot_fs_));
+    test_db_env_.reset(new CompositeEnvWrapper(Env::Default(), test_db_fs_));
+    options_.env = test_db_env_.get();
+    test_backup_env_.reset(
+        new CompositeEnvWrapper(Env::Default(), test_backup_fs_));
+    if (engine_options_) {
+      engine_options_->backup_env = test_backup_env_.get();
+    }
+    file_manager_.reset(new FileManager(backup_chroot_env_.get()));
+    db_file_manager_.reset(new FileManager(db_chroot_env_.get()));
+
+    // Create logger
+    DBOptions logger_options;
+    logger_options.env = db_chroot_env_.get();
+    // TODO: This should really be an EXPECT_OK, but this CreateLogger fails
+    // regularly in some environments with "no such directory"
+    CreateLoggerFromOptions(dbname_, logger_options, &logger_)
```

Comment:
I'm getting the following exception one macos, the `logger_` is not a valid Logger, could it because it's not created here?:
<img width="2859" alt="image" src="https://user-images.githubusercontent.com/33968/156679366-0ca4c150-ef72-4442-a8b7-17fc5303410e.png">

Reply:
Not sure. I fixed the non-OK issue on Linux. See if that works.

Reply:
Fix a UAF based on ASAN trace so should be fixed now.